pkg/prometheus/resource_selector.go

```diff
@@ -1466,3 +1471,37 @@ func (rs *ResourceSelector) validateScalewaySDConfigs(ctx context.Context, sc *m
 	}
 	return nil
 }
+
+func (rs *ResourceSelector) validateIonosSDConfigs(ctx context.Context, sc *monitoringv1alpha1.ScrapeConfig) error {
+	promVersion := operator.StringValOrDefault(rs.p.GetCommonPrometheusFields().Version, operator.DefaultPrometheusVersion)
+	version, err := semver.ParseTolerant(promVersion)
+	if err != nil {
+		return fmt.Errorf("failed to parse Prometheus version: %w", err)
+	}
+	if !version.GTE(semver.MustParse("2.36.0")) {
+		return fmt.Errorf("ionos SD configuration is only supported for Prometheus version >= 2.36.0")
+	}
+
+	for i, config := range sc.Spec.IonosSDConfigs {
+		if err := rs.store.AddBasicAuth(ctx, sc.GetNamespace(), config.BasicAuth); err != nil {
```

Comment:
One of the `BasicAuth`, `Authorization`, `OAuth2`  must be required?

Reply:
I cannot see any evidence for this in the Prometheus Code, do you mean to say we should validate that at least one of: `BasicAuth`, `Authorization`, or `OAuth2` is configured?

Reply:
Hmmm.... External cloud service APIs must be authenticated. Maybe we can optimize it uniformly. 🤔️

Reply:
We are thinking of using CEL to authenticate that the three fields are mutually exclusive. Maybe we can also use CEL to ensure that atleast one of them is present too. This is just speculation though, I am still not sure if that is something we want to enforce. 
Maybe @simonpasquier (for when he is back) and @slashpai have a better idea.

Reply:
https://docs.ionos.com/reference/readme/get-started#authorization says that bearer token authentication is preferred (basic-auth is supported too but deprecated). Since it's a new SD, we could just support authorization here as it simplifies the work.